priv/apps/itest/lib/in_flight_exit_client.ex

```diff
@@ -0,0 +1,262 @@
+    :exit_id,
+    :piggyback_input_hash,
+    :process_exit_receipt_hash,
+    :receiver_address,
+    :sign_hash,
+    :signed_txbytes,
+    :start_in_flight_exit,
+    :txbytes
+  ]
+
+  def start_in_flight_exit(sender_address, sender_key, receiver_address) do
+    %__MODULE__{address: sender_address, address_key: sender_key, receiver_address: receiver_address}
+    |> create_transaction()
+    |> sign_transaction()
+    |> add_exit_queue()
+    |> get_exit_data()
+    |> get_exit_game_contract_address()
+    |> do_in_flight_exit()
+    |> get_in_flight_exit_id()
+    |> piggyback_input()
+    |> process_exit()
+  end
+
+  defp get_exit_data(%{signed_txbytes: txbytes} = se) do
+    payload = %InFlightExitTxBytesBodySchema{txbytes: txbytes}
+    {:ok, response} = InFlightExit.in_flight_exit_get_data(Watcher.new(), payload)
+
+    exit_data = Jason.decode!(response.body)["data"]
+    %{se | exit_data: exit_data}
+  end
+
+  defp create_transaction(%{address: address, receiver_address: receiver_address} = se) do
+    {:ok, [sign_hash, _typed_data, txbytes]} =
+      Client.create_transaction(Currency.to_wei(1), address, receiver_address, Currency.ether())
+
+    %{se | txbytes: txbytes, sign_hash: sign_hash}
+  end
+
+  defp sign_transaction(%{address_key: address_key, txbytes: txbytes, sign_hash: sign_hash} = se) do
+    [type, inputs, outputs, metadata] = txbytes |> Encoding.to_binary() |> ExRLP.decode()
+
+    signature = sign_hash |> Encoding.to_binary() |> Encoding.signature_digest(address_key)
+
+    rlp_list = [[signature], type, inputs, outputs, metadata]
+
+    signed_txbytes = rlp_list |> ExRLP.encode() |> Encoding.to_hex()
+
+    %{se | signed_txbytes: signed_txbytes}
+  end
+
+  defp get_exit_game_contract_address(se) do
+    data = ABI.encode("exitGames(uint256)", [PaymentType.simple_payment_transaction()])
+    {:ok, result} = Ethereumex.HttpClient.eth_call(%{to: Itest.Account.plasma_framework(), data: Encoding.to_hex(data)})
+
+    exit_game_contract_address =
+      result
+      |> Encoding.to_binary()
+      |> ABI.TypeDecoder.decode([:address])
+      |> hd()
+      |> Encoding.to_hex()
+
+    %{se | exit_game_contract_address: exit_game_contract_address}
+  end
+
+  defp do_in_flight_exit(
+         %{address: address, exit_data: exit_data, exit_game_contract_address: exit_game_contract_address} = se
+       ) do
+    in_flight_tx = exit_data["in_flight_tx"] |> Encoding.to_binary()
+    in_flight_tx_sigs = Enum.map(exit_data["in_flight_tx_sigs"], &Encoding.to_binary(&1))
+    input_txs = Enum.map(exit_data["input_txs"], &Encoding.to_binary(&1))
+    input_txs_inclusion_proofs = Enum.map(exit_data["input_txs_inclusion_proofs"], &Encoding.to_binary(&1))
+    input_utxos_pos = Enum.map(exit_data["input_utxos_pos"], &:binary.encode_unsigned(&1))
+
+    # NOTE: hardcoded for now, we're talking to a particular exit game so this is fixed
+    optional_bytes_array = List.duplicate("", Enum.count(input_txs))
```

Comment:
what is hardcoded? why?

Reply:
This is pulled behavior from `OMG.Eth`. I can add a comment to explain that if it's clearer.

Reply:
yeah, or ... make it better. this doesn't give me assurance that "hey, this is really part of the protocol we're implementing". it looks more like we had to do this because we're handling data wrong. don't you think?